src/More/LanguageSelection.tsx

```diff
@@ -1,71 +1,139 @@
-import React from "react"
-import {
-  FlatList,
-  View,
-  StyleSheet,
-  TouchableHighlight,
-  SafeAreaView,
-} from "react-native"
+import React, { FunctionComponent } from "react"
+import { FlatList, View, StyleSheet, TouchableOpacity } from "react-native"
 import { useTranslation } from "react-i18next"
 import { useNavigation } from "@react-navigation/native"
 import { useStatusBarEffect } from "../navigation"
+import { SvgXml } from "react-native-svg"
 
 import { getLocaleList, setUserLocaleOverride } from "../locales/languages"
 import { GlobalText } from "../components/GlobalText"
+import { Icons } from "../assets"
 
-import { Colors, Spacing, Typography } from "../styles"
-
-const Separator = () => (
-  <View
-    style={{
-      backgroundColor: Colors.primaryBorder,
-      height: StyleSheet.hairlineWidth,
-      width: "100%",
-    }}
-  />
-)
+import {
+  Outlines,
+  Layout,
+  Iconography,
+  Colors,
+  Spacing,
+  Typography,
+} from "../styles"
 
-const LanguageSelection = (): JSX.Element => {
+const LanguageSelection: FunctionComponent = () => {
   const {
     i18n: { language },
+    t,
   } = useTranslation()
   const navigation = useNavigation()
-  useStatusBarEffect("dark-content")
+  useStatusBarEffect("light-content")
   const localeList = getLocaleList()
 
-  const onSelectLanguage = (locale: string) => {
-    setUserLocaleOverride(locale)
-    navigation.goBack()
+  type ListItem = {
+    label: string
+    value: string
+  }
+
+  const renderItem = ({ item }: { item: ListItem }) => {
+    const handleOnSelectLanguage = () => {
+      setUserLocaleOverride(item.value)
+      navigation.goBack()
+    }
+
+    const languageIsSelected = language === item.value
+    const languageButtonTextStyle = languageIsSelected
+      ? { ...Typography.bold, color: Colors.primaryText }
+      : null
+    const languageButtonTextStyles = {
+      ...style.languageButtonText,
+      ...languageButtonTextStyle,
+    }
+
+    return (
+      <TouchableOpacity
+        style={style.languageButton}
+        onPress={handleOnSelectLanguage}
+      >
+        <GlobalText style={languageButtonTextStyles}>{item.label}</GlobalText>
+      </TouchableOpacity>
+    )
   }
 
+  const headerHeight = 70
```

Comment:
should these styles live in the stylesheet?

Reply:
Oh, yeah the should. For some reason I was getting an error when trying to do that before, but looks good now.